integrationtest/core/src/main/scala/io/gearpump/integrationtest/minicluster/RestClient.scala

```diff
@@ -75,7 +75,11 @@ class RestClient(host: String, port: Int) {
     listApps().filter(_.status == MasterToAppMaster.AppMasterActive)
   }
 
-  def submitApp(jar: String, args: String = "", config: String = ""): Int = try {
+  def getNextAvailableAppId(): Int = {
+    listApps().length + 1
```

Comment:
Your version looks better. as long as `listApps` does not return an empty Array. Actually `listApps` will return all apps including killed. `listRunningApps` will only return running apps, which we cannot use.